utils/src/main/kotlin/OkHttpClientHelper.kt

```diff
@@ -69,17 +69,23 @@ object OkHttpClientHelper {
      * Apply HTTP proxy settings from a [url], optionally with credentials included.
      */
     fun OkHttpClient.Builder.applyProxySettingsFromUrl(url: URL) {
-        proxy(Proxy(Proxy.Type.HTTP, InetSocketAddress(url.host, url.port)))
-        proxyAuthenticator(object : Authenticator {
-            override fun authenticate(route: Route?, response: Response): Request? {
-                val user = url.userInfo.substringBefore(':')
-                val password = url.userInfo.substringAfter(':')
-                val credential = Credentials.basic(user, password)
-                return response.request.newBuilder()
-                    .header("Proxy-Authorization", credential)
-                    .build()
+        if (url.host.isNotEmpty()) {
```

Comment:
How about reducing the amount of different indentation levels by starting the function with:

if (url.host.isEmpty() || url.userInfo==null) return

Reply:
But then we'd still need further conditionals to find out later which of the two was not the case, and whether to call only `proxy()` or both `proxy()` and `proxyAuthenticator()`.

Reply:
But I got your point and did it slightly differently.